### ISSUE TITLE
"float64 size isn't 8 bytes on this platform", can I ignore?

### ISSUE THREAD

**Issue by Kazade** [g-truc/glm#627]
I'm trying to compile a library of mine (which uses GLM) for the Sega Dreamcast, and because I'm compiling to an SH4 CPU, I hit this error here:

```
type_float.hpp:62:2: error: static assertion failed: float64 size isn't 8 bytes on this platform
```

My question is, am I OK to ignore this / comment out the assertion? Does GLM rely on this being 8 bytes elsewhere? I should be OK if I just don't use float64 right?

Thanks!

**Comment by Groovounet** [comment]
Hi
What compiler are you using? Does it mean there is no double on your platform? I never really consider this case... It's probably not a big deal but I guess I'll like to figure something smoother for this platform.

**Comment by Kazade** [comment]
The compiler is GCC 4.7.3 although it's patched for the Dreamcast. I'm guessing that double is the same size as float on this platform.

**Comment by Groovounet** [comment]
Can you confirm that sizeof(double) is 4 on your Dreamcast?

**Comment by Kazade** [comment]
Yes confirmed. Double and float are the same size (it's compiled with --m4-single-only)

**Comment by Groovounet** [comment]
This has been implement using the define GLM_FORCE_SINGLE_ONLY.

When defined before including GLM, all requirements for 64-bits float is released.

This feature is implemented in master branch for GLM 0.9.9 release.

Thanks for reporting,
Christophe
